pkg/backup/backup_compaction.go

```diff
@@ -406,50 +424,74 @@ func openSSTs(
 	}, nil
 }
 
-// makeCompactionBackupDetails takes a chain of backups that are to be
-// compacted and returns a corresponding BackupDetails for the compacted
-// backup. It also takes in the job details for the last backup in its chain.
+// makeCompactionBackupDetails takes a backup chain (up until the end timestamp)
+// and returns a corresponding BackupDetails for the compacted
+// backup of backups from the start timestamp to the end timestamp.
 func makeCompactionBackupDetails(
 	ctx context.Context,
-	lastIncDetails jobspb.BackupDetails,
-	manifests []backuppb.BackupManifest,
-	dest backupdest.ResolvedDestination,
-	encryptionOptions *jobspb.BackupEncryptionOptions,
+	compactionChain compactionChain,
+	dest jobspb.BackupDetails_Destination,
+	resolvedDest backupdest.ResolvedDestination,
+	encryption *jobspb.BackupEncryptionOptions,
 	kmsEnv cloud.KMSEnv,
 ) (jobspb.BackupDetails, error) {
-	if len(manifests) == 0 {
+	if len(compactionChain.chainToCompact) == 0 {
 		return jobspb.BackupDetails{}, errors.New("no backup manifests to compact")
 	}
-	compactedDetails, err := updateBackupDetails(
-		ctx, lastIncDetails,
-		dest.CollectionURI,
-		dest.DefaultURI,
-		dest.ChosenSubdir,
-		dest.URIsByLocalityKV,
-		manifests,
-		encryptionOptions,
-		kmsEnv,
+	var encryptionInfo *jobspb.EncryptionInfo
+	if encryption != nil {
+		var err error
+		_, encryptionInfo, err = backupencryption.MakeNewEncryptionOptions(
+			ctx,
+			*encryption,
+			kmsEnv,
+		)
+		if err != nil {
+			return jobspb.BackupDetails{}, err
+		}
+	}
+	lastBackup := compactionChain.lastBackup()
+	allDescs, _, err := backupinfo.LoadSQLDescsFromBackupsAtTime(
+		ctx,
+		compactionChain.chainToCompact,
+		compactionChain.compactedIterFactory,
+		compactionChain.end,
 	)
 	if err != nil {
 		return jobspb.BackupDetails{}, err
 	}
-	// The manifest returned by updateBackupDetails have its start and end times
-	// set to append to the chain. We need to update them to reflect the
-	// compacted backup's start and end times.
-	compactedDetails.StartTime = manifests[0].StartTime
-	compactedDetails.EndTime = manifests[len(manifests)-1].EndTime
+	allDescsPb := util.Map(allDescs, func(desc catalog.Descriptor) descpb.Descriptor {
+		return *desc.DescriptorProto()
+	})
+	compactedDetails := jobspb.BackupDetails{
```

Comment:
after we jobify this: we should think about how to write a test that ensures that anyone who ads a field to backup job details remembers to populate it on the conventional backup path, and here.